### ISSUE TITLE
secret id is not always canonicalized before calling secret_get

### ISSUE THREAD

**Issue by PietroPasotti** [canonical/operator#902]
Secret canonicalizes the ID it receives upon initialization.
However, before the Secret object is initialized, the ID is used to call secret_get. This results in the ID being sometimes canonicalized, sometimes not, depending on the caller.

For example:
```
self.model.get_secret(id="foo").get_content()
# resulting call: secret-get --id foo

self.model.get_secret(id="foo").peek_content()
# resulting call: secret-get --id secret:foo --peek
```

This is not a problem per se, but it might mean headache in the future.

**Comment by PietroPasotti** [comment]
Proposal:
move the _canonicalize_id call out of Secret and into ModelBackend.

```
    def get_secret(self, *, id: Optional[str] = None, label: Optional[str] = None) -> 'Secret':
        if not (id or label):
            raise TypeError('Must provide an id or label, or both')
        id = Secret._canonicalize_id(id) if id else None    
        
        try:
            content = self._backend.secret_get(id=id, label=label)
            return Secret(self._backend, id=id, label=label, content=content)
        except ModelError:
            # TODO(benhoyt): remove the secret-info-get fallback once
            #  juju/juju#14916 is fixed (should be in Juju 3.0.3)
            info = self._backend.secret_info_get(id=id, label=label)
            return Secret(self._backend, id=info.id, label=info.label)
```

In Secret, we might put a check that verifies that the ID it is passed is already canonicalized.

**Comment by benhoyt** [comment]
@PietroPasotti Fixed by #903 -- please review.
